app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt

```diff
@@ -288,6 +299,22 @@ public class MainActivity : AppCompatActivity(), MainViewController, RouteCallba
         mapController?.mapTilt = radians
     }
 
+    override fun toggleMute() {
+        //Update cached value of routePresenter.isMuted
+        val routePresenter = routeModeView.routePresenter
+        routePresenter?.onMuteClicked()
+
+        val muted = (routePresenter?.isMuted() == true)
+        muteView.setMuted(!muted)
+
+        //Actually mute or unmute the speakerbox based on current value
+        if (muted) {
+            routeModeView.voiceNavigationController!!.mute()
```

Comment:
Prefer `?.` over `!!.` when referencing nullable variables. Kotlin makes it harder to get NPEs but if you really want one `!!.` is one of the best way to get them ;)

https://kotlinlang.org/docs/reference/null-safety.html